commands/addons/create.js

```diff
@@ -45,23 +45,28 @@ function * run (context, heroku) {
   const util = require('../../lib/util')
 
   let {app, flags, args} = context
-  let {name, as, confirm} = flags
+  let {name, as} = flags
   let plan = {name: args[0]}
   let config = parseConfig(args.slice(1))
 
-  let addon
-
-  yield cli.action(`Creating ${plan.name} on ${cli.color.app(app)}`, co(function * () {
-    addon = yield heroku.post(`/apps/${app}/addons`, {
+  function createAddon (app, config, name, confirm, plan, as) {
+    cli.action.start(`Creating ${plan.name} on ${cli.color.app(app)}`)
+    return heroku.post(`/apps/${app}/addons`, {
       body: { config, name, confirm, plan, attachment: {name: as} },
       headers: {
         'accept-expansion': 'plan',
         'x-heroku-legacy-provider-messages': 'true'
       }
+    }).catch(function (err) {
+      cli.action.done(cli.color.red.bold('!'))
```

Comment:
@dickeyxxx yeah, I fixed it in 8b6399a, I had things arranged differently before, and refactored to this without going back the previous action usage